Automattic/wp-calypso
# Stepper: Do not clear flowName in step-route tracker

Fixes https://github.com/Automattic/wp-calypso/issues/97077

## Proposed Changes

This keeps the flow name persisted after reading it in step-route-tracking hook. This hook assumed it was the only part of the code that relied on this value and this deleted it. This corrects that assumption.

## Why are these changes being made?
To prevent sites from being recreated when coming back from checkout to domains step.

## Testing Instructions

1. Go to /setup/onboarding.
2. Make a site with a paid domain.
3. When you land at checkout, note the siteSlug param.
4. Go back to the domains step and continue to checkout again.
5. The site slug should be the same.

## Regression testing
@daledupreez would really appreciate it if you can take a look. I'm curious to find out if you picked the `andClear` variants for a reason.
